tests/test_models/phelps_deaton.py

```diff
@@ -1,65 +1,71 @@
-"""Define example model specifications."""
+"""Example specifications of the Phelps-Deaton model."""
 
 import jax.numpy as jnp
 
-RETIREMENT_AGE = 65
-N_CHOICE_GRID_POINTS = 500
-N_STATE_GRID_POINTS = 100
+# ======================================================================================
+# Numerical parameters and constants
+# ======================================================================================
 
+N_GRID_POINTS = {
+    "states": 100,
+    "choices": 500,
+}
 
-def phelps_deaton_utility_with_shock(
-    consumption,
-    working,
-    delta,
-    additive_utility_shock,
-):
-    return jnp.log(consumption) + additive_utility_shock - delta * working
+RETIREMENT_AGE = 65
 
+# ======================================================================================
+# Model functions
+# ======================================================================================
 
-def phelps_deaton_utility(consumption, working, delta):
+
+# --------------------------------------------------------------------------------------
+# Utility functions
+# --------------------------------------------------------------------------------------
+def utility(consumption, working, delta):
     return jnp.log(consumption) - delta * working
 
 
-def phelps_deaton_utility_with_filter(
+def utility_with_filter(
     consumption,
     working,
     delta,
     lagged_retirement,  # noqa: ARG001
 ):
-    return jnp.log(consumption) - delta * working
+    return utility(consumption=consumption, working=working, delta=delta)
 
 
+# --------------------------------------------------------------------------------------
+# Auxiliary variables
+# --------------------------------------------------------------------------------------
 def working(retirement):
     return 1 - retirement
 
 
-def next_wealth_with_shock(
-    wealth,
-    consumption,
-    working,
-    wage,
-    wage_shock,
-    interest_rate,
-):
-    return interest_rate * (wealth - consumption) + wage * wage_shock * working
+def wage(age):
+    return 1 + 0.1 * age
+
+
+def age(_period):
+    return _period + 18
 
 
+# --------------------------------------------------------------------------------------
+# State transitions
+# --------------------------------------------------------------------------------------
 def next_wealth(wealth, consumption, working, wage, interest_rate):
     return (1 + interest_rate) * (wealth - consumption) + wage * working
```

Comment:
See above, this is really weird. Funny that I notice only now. Can we add a reference where we got the model from? Was that the DC-EGM paper?

Reply:
![image](https://github.com/OpenSourceEconomics/lcm/assets/39708667/0e85d417-c7cf-4f1f-a386-26ec1a5e49d4)

It is from the DC-EGM model. I will add a reference.